packages/plugin-commands-publishing/src/gitChecks.ts

```diff
@@ -0,0 +1,47 @@
+import PnpmError from '@pnpm/error'
+import execa = require('execa')
+
+// git checks logic is from https://github.com/sindresorhus/np/blob/master/source/git-tasks.js
+
+export async function gitChecks () {
+  if (await getCurrentBranch() !== 'master') {
```

Comment:
I am not sure about this check. What if I want to publish a prerelease version (like 2.0.0-beta.0) from a branch.

cc @pnpm/collaborators

Reply:
Maybe instead of a failure we should show a y/N prompt, so that the user can confirm that he wants to publish from a different branch.